rero_ils/modules/utils.py

```diff
@@ -436,23 +436,25 @@ def pids_exists_in_data_test(info, data, tests, is_required):
         return_value = []
         endpoints = current_app.config['RECORDS_REST_ENDPOINTS']
         for pid_type, keys in tests.items():
+            # make a list of keys
             if isinstance(keys, str):
                 keys = [keys]
             for key in keys:
-                data_to_test = data.get(key)
-                if data_to_test:
+                data_to_test_list = data.get(key, [])
+                if isinstance(data_to_test_list, dict):
+                    data_to_test_list = [data_to_test_list]
+                for data_to_test in data_to_test_list:
                     try:
                         list_route = endpoints[pid_type]['list_route']
                         data_pid = data_to_test.get('pid') or \
                             data_to_test.get('$ref').split(list_route)[1]
-                    except Exception:
+                    except Exception as err:
```

Comment:
if `err` is never used, why assign it ? It will provide useless work for garbage collector :-)